### DISCUSSION TITLE
Help Connecting Container to Gluetun

### DISCUSSION THREAD

**Discussion by djphllps** [qdm12/gluetun#581] [Q&A] [answered]
Hello, I'm pretty new to docker so forgive me for any dumbness that is about to ensue pls.   I have created a Gluetun Container on my QNAP container station with ...

```
docker run -d --name gluetun \
  --cap-add=NET_ADMIN \
 -e VPNSP="Cyberghost" \
 -e OPENVPN_USER=Blah \
 -e OPENVPN_PASSWORD=Blah \
 -e CYBERGHOST_GROUP="Premium UDP USA" \
 -v "/share/CACHEDEV1_DATA/Atlas Shared/Gluetun Config":/gluetun \
 qmcgaw/gluetun
```

i then created a qbittorrent container with

```
docker run -d --name=qbittorrent --network=container:gluetun \
  -e PUID=1000 \
  -e PGID=100 \
  -e TZ=Eastern \
  -e WEBUI_PORT=8080 \
  -p 6881:6881 \
  -p 6881:6881/udp \
  -p 8080:8080 \
  -v "/share/CACHEDEV1_DATA/Atlas Shared/Downloads/Qbittorrent Config":/config \
  -v "/share/CACHEDEV1_DATA/Atlas Shared/Downloads":/downloads \
  --restart unless-stopped \
  ghcr.io/linuxserver/qbittorrent
```

however for the above run command to work i did have to remove the port environmental variables because i kept getting an error .  So i guess at this point what i need to know is what adjustments do i need to make to these commands to get the qbittorrent container working VPN'd by the gluetun container but still accessible via a webui port so myself and sonarr can use it.  I did see a docker compose instruction on how to do this but that is different than the way i have figured out how to do things i don't really know how to do docker compose.  Thanks in advance.  Oh one more thing i also noticed that my VPN kept connecting in Chile even though i put USA as the server group ...could use help with that too but its not critical since its technically connecting.

**Comment by qdm12** [comment, marked as answer]
Hi there!

> what adjustments do i need to make to these commands to get the qbittorrent container working VPN'd by the gluetun container but still accessible via a webui port so myself and sonarr can use it.

You need to publish the web ui port (say `8080`) on gluetun instead, so add `-p 8080:8080` to the gluetun run command. Don't publish any port on your qbittorrent run command. This is because qbittorrent uses gluetun as its network host, so gluetun has to handle the networking part.

> Oh one more thing i also noticed that my VPN kept connecting in Chile even though i put USA as the server group ...could use help with that too but its not critical since its technically connecting.

That's because Cyberghost `Premium UDP USA` includes south america as well ~I think~ (I'm certain, see [here](https://github.com/qdm12/gluetun/wiki/Cyberghost-Servers)). I believe you can just remove the `CYBERGHOST_GROUP` and set the ~`-e COUNTRY=USA` (or whatever corresponds to usa, it will complain if it's not valid)~ `-e REGION="United States"` (see [the list of servers](https://github.com/qdm12/gluetun/wiki/Cyberghost-Servers))

**Reply by djphllps** [reply]
OMG ty so much i was obviously over thinking this.  I really appreciate your time.

**Comment by ChrizZz90** [comment]
Is there a smart way to link gluetun as network_mode in Portainer? I tried editing my containers and selected container -> gluetun but portainer saves this based on the container ID and after every rebuild/restart it crashes all other containers. 

Maybe as a 2nd topic: Can I link containers that they start after gluetun is started or do I have to create 1 compose file to get this as expected?

**Reply by qdm12** [reply]
Subscribe to releases for github.com/qdm12/deunhealth I'll add a solution to restart and reconnect containers to gluetun once they lose connection. For now it only restarts unhealthy containers but this doesn't work if gluetun is updated & restarted.

Personally for now, all my gluetun containers are in the same docker-compose.yml and I do `docker-compose down && docker-compose pull && docker-compose up -d` to update and restart all of them.

AFAIK, there is no other simple way to do it. Maybe with some host ip routing/iptables but I don't really want to explore that route and mess with the host.

**Reply by ChrizZz90** [reply]
great, I subscribed to your deunhealth project. I am using Watchtower which checks all containers on a daily basis and updates them. Would be great to keep the comfort
